src/main/java/macrobase/analysis/BatchAnalyzer.java

```diff
@@ -63,87 +60,19 @@ public AnalysisResult analyze() throws SQLException, IOException, ConfigurationE
-
-        log.debug("...ended loading (time: {}ms)!", loadTime);
-
-        Stopwatch tsw = Stopwatch.createUnstarted();
-        Stopwatch tsw2 = Stopwatch.createUnstarted();
-        tsw.start();
-        tsw2.start();
-
-        sw.start();
-
-        OutlierDetector detector = constructDetector(randomSeed);
-
-        OutlierDetector.BatchResult or;
-        if (forceUsePercentile || (!forceUseZScore && targetPercentile > 0)) {
-            or = detector.classifyBatchByPercentile(data, targetPercentile);
-        } else {
-            or = detector.classifyBatchByZScoreEquivalent(data, zScore);
-        }
-        sw.stop();
-
-        long classifyTime = sw.elapsed(TimeUnit.MILLISECONDS);
-        sw.reset();
-
-        tsw2.stop();
-
-        // STORE RESULTS IF SPECIFIED
-
-        if (this.storeAnalysisResults != null) {
-            this.storeAnalysisResultsInJson(or);
-        }
-
-        // SUMMARY
-
-        final int inlierSize = or.getInliers().size();
-        final int outlierSize = or.getOutliers().size();
-
-        log.debug("Starting summarization...");
-
-        sw.start();
-        FPGrowthEmerging fpg = new FPGrowthEmerging();
-        List<ItemsetResult> isr = fpg.getEmergingItemsetsWithMinSupport(or.getInliers(),
-                                                                        or.getOutliers(),
-                                                                        minSupport,
-                                                                        minOIRatio,
-                                                                        encoder);
-        sw.stop();
-        tsw.stop();
-
-        double tuplesPerSecond = ((double) data.size()) / ((double) tsw.elapsed(TimeUnit.MICROSECONDS));
-        tuplesPerSecond *= 1000000;
+        DataIngester ingester = constructIngester();
+        FeatureTransform featureTransform = new BatchFeatureTransform(conf, ingester, conf.getDetectorType());
 
-        long summarizeTime = sw.elapsed(TimeUnit.MILLISECONDS);
-        sw.reset();
-        log.debug("...ended summarization (time: {}ms)!", summarizeTime);
+        OutlierClassifier classifier = new BatchingPercentileClassifier(conf, featureTransform);
 
-        log.debug("Number of itemsets: {}", isr.size());
-        log.debug("...ended total (time: {}ms)!", (tsw.elapsed(TimeUnit.MICROSECONDS) / 1000) + 1);
-        log.debug("Tuples / second = {} tuples / second", tuplesPerSecond);
+        Summarizer summarizer = new BatchSummarizer(conf, classifier);
 
-        tuplesPerSecond = ((double) data.size()) / ((double) tsw2.elapsed(TimeUnit.MICROSECONDS));
-        tuplesPerSecond *= 1000000;
-        log.debug("Tuples / second w/o itemset mining = {} tuples / second", tuplesPerSecond);
+        Summary result = summarizer.next();
 
-        return new AnalysisResult(outlierSize, inlierSize, loadTime, classifyTime, summarizeTime, isr);
+        return new AnalysisResult(0, 0, 0, 0, 0, result.getItemsets());
```

Comment:
Would be nice to have a TODO here to make sure we eventually don't just return 0s. (Better yet, some code to actually return the right values)